src/seedu/addressbook/logic/Logic.java

```diff
@@ -66,15 +68,30 @@ protected void setLastShownList(List<? extends ReadOnlyPerson> newList) {
 
     /**
      * Parses the user command, executes it, and returns the result.
+     * Record this user command.
      * @throws Exception if there was any problem during command execution.
      */
     public CommandResult execute(String userCommandText) throws Exception {
+        userCommandText = checkLastCommand(userCommandText);
         Command command = new Parser().parseCommand(userCommandText);
         CommandResult result = execute(command);
         recordResult(result);
         return result;
     }
 
+    /**
+     * Check if the command is "last".
+     * If yes then append the lastCommand for later execution.
+     */
+    private String checkLastCommand(String userCommandText) throws Exception {
+        if( userCommandText.substring(0, Math.min(userCommandText.length(), 4)).equals("last")) {
+            userCommandText = userCommandText.concat(" ").concat(lastCommand);
```

Comment:
It seems this is logically wrong. if the user type "last" and the previous command is "add something", then you concatenate both here before parsing. But the parser will not understand the concatenated string as it is not a valid command. I hope this process should be carried out once you parse the command (the application realizes it is the command LAST).